lib/fluent/plugin/in_prometheus.rb

```diff
@@ -32,6 +33,9 @@ class PrometheusInput < Fluent::Plugin::Input
       config_param :extra_conf, :hash, default: nil, symbolize_keys: true, deprecated: 'See http helper config'
     end
 
+    desc 'Content encoding of the exposed metrics, Currently supported encoding is identity, gzip. Ref: https://prometheus.io/docs/instrumenting/exposition_formats/#basic-info'
+    config_param :content_encoding, :string, default: "identity"
```

Comment:
```suggestion
    config_param :content_encoding, :enum, list: [:identify, :gzip], default: :identify
```

It would be better to use the config value type :enum instead of :string to specify fixed choices, so that following check by plugin isn't needed.

Reply:
One doubt here
Does the list have to be list of symbols?

Reply:
Yes, list items have to be symbol:
https://github.com/fluent/fluentd/blob/64595e2ac291a3e1d2d02feefdc58e623b3abb02/lib/fluent/config/types.rb#L126